src/main/java/it/eng/idsa/businesslogic/routes/CamelRouteReceiver.java

```diff
@@ -31,114 +33,149 @@
 
 @Component
 public class CamelRouteReceiver extends RouteBuilder {
-	
+
 	private static final Logger logger = LogManager.getLogger(CamelRouteReceiver.class);
-	
+
 	@Autowired
 	private ApplicationConfiguration configuration;
 
 	@Autowired(required = false)
 	ReceiverFileRecreatorProcessor fileRecreatorProcessor;
-	
+
 	@Autowired
 	ReceiverParseReceivedConnectorRequestProcessor connectorRequestProcessor;
 
 	@Autowired
 	ValidateTokenProcessor validateTokenProcessor;
-	
+
 	@Autowired
 	ContractAgreementProcessor contractAgreementProcessor;
 	
 	@Autowired
 	ReceiverMultiPartMessageProcessor multiPartMessageProcessor;
-	
+
 	@Autowired
 	ReceiverSendDataToDataAppProcessor sendDataToDataAppProcessor;
-	
+
 	@Autowired
 	RegisterTransactionToCHProcessor registerTransactionToCHProcessor;
-	
+
 	@Autowired
 	ExceptionProcessorReceiver exceptionProcessorReceiver;
-	
+
 	@Autowired
 	GetTokenFromDapsProcessor getTokenFromDapsProcessor;
-	
+
 	@Autowired
 	ReceiverSendDataToBusinessLogicProcessor sendDataToBusinessLogicProcessor;
-	
+
 	@Autowired
 	ReceiverWebSocketSendDataToDataAppProcessor sendDataToDataAppProcessorOverWS;
 
 	@Autowired
 	ReceiverUsageControlProcessor receiverUsageControlProcessor;
+	
+	@Autowired
+	MapIDSCP2toMultipart mapIDSCP2toMultipart;
 
 	@Autowired
-	CamelContext camelContext;
+	MapMultipartToIDSCP2 mapMultipartToIDSCP2;
 
-	@Value("${application.idscp.isEnabled}")
-	private boolean isEnabledIdscp;
+	@Autowired
+	CamelContext camelContext;
 
 	@Value("${application.websocket.isEnabled}")
 	private boolean isEnabledWebSocket;
 
+	@Value("${application.idscp2.isEnabled}")
+	private boolean isEnabledIdscp2;
+
+	@Value("${application.isReceiver}")
+	private boolean receiver;
+
+	@Value("${application.dataApp.websocket.isEnabled}")
+	private boolean isEnabledDataAppWebSocket;
+
 	@SuppressWarnings("unchecked")
 	@Override
 	public void configure() throws Exception {
 		logger.debug("Starting Camel Routes...receiver side");
-        camelContext.getShutdownStrategy().setLogInflightExchangesOnTimeout(false);
-        camelContext.getShutdownStrategy().setTimeout(3);
+		camelContext.getShutdownStrategy().setLogInflightExchangesOnTimeout(false);
+		camelContext.getShutdownStrategy().setTimeout(3);
 
-      //@formatter:off
+		//@formatter:off
 		onException(ExceptionForProcessor.class, RuntimeException.class)
 			.handled(true)
 			.process(exceptionProcessorReceiver)
 			.process(getTokenFromDapsProcessor)
 			.process(registerTransactionToCHProcessor)
 			.process(sendDataToBusinessLogicProcessor);
-
-		// Camel SSL - Endpoint: B
-		if(!isEnabledIdscp && !isEnabledWebSocket) {
+		
+		// Camel SSL - Endpoint: B communication http
+		if(!isEnabledDataAppWebSocket && !isEnabledWebSocket) {
```

Comment:
What happens if dataApp is REST and ECC is WSS?
Where will that case fall in?
Or the other way: ECC is REST and dataApp is WSS